pkg/controller/elasticsearch/certificates/transport/reconcile.go

```diff
@@ -104,7 +106,7 @@ func ReconcileTransportCertificatesSecrets(
 	}
 
 	if !reflect.DeepEqual(secret, currentTransportCertificatesSecret) {
-		if err := c.Update(&secret); err != nil {
+		if err := c.Update(secret); err != nil {
```

Comment:
I think this is the place where we could potentially benefit from the generic secret reconciler, but it requires a bit more work. I'll open a follow-up issue.